internal/parser/reparser.go

```diff
@@ -44,6 +44,10 @@ func (p *Parser) reparseTags(parent *ast.Node, jsDoc []*ast.Node) {
 			continue
 		}
 		for _, tag := range tags.Nodes {
+			if parent.Kind == ast.KindEndOfFile {
+				p.reparseUnhosted(tag, parent, j)
+				continue
+			}
```

Comment:
this seems to make sense to me that htis should be treated as unhosted JSDoc. But despite this... issues like this did not go away (yet): https://github.com/microsoft/typescript-go/pull/1187#discussion_r2146155726

I need to investigate further, cc @sandersn